test/data/testrepos/appstream/repodata/repomd.xml

```diff
@@ -0,0 +1,80 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<repomd xmlns="http://linux.duke.edu/metadata/repo" xmlns:rpm="http://linux.duke.edu/metadata/rpm">
+  <revision>9-stream</revision>
+  <tags>
+    <distro cpeid="cpe:/o:centos-stream:centos-stream:9">CentOS Stream 9</distro>
+  </tags>
+  <data type="primary">
+    <checksum type="sha256">c03bb2ebc33bbee3046da091b45d90717273a43d3641a9d84e6eede05637bc8a</checksum>
+    <open-checksum type="sha256">a1df0d610cd466c0ff1638782c5a1427b19a01bcde147105e9a07f87669f038a</open-checksum>
+    <location href="repodata/c03bb2ebc33bbee3046da091b45d90717273a43d3641a9d84e6eede05637bc8a-primary.xml.gz"/>
+    <timestamp>1736761560</timestamp>
+    <size>3597627</size>
+    <open-size>51840420</open-size>
+  </data>
+  <data type="filelists">
+    <checksum type="sha256">72d4b6933f70291d407371a5ebc4cdb26b3e0ea37b176657a8f3e4d40d9f32f6</checksum>
```

Comment:
Silly question, could we just remove the filelist from the repomd.xml and also from this PR somehow? Or play some tricks to achieve that?  It makes me sad that we need to add 17Mb of binary data to our git repo that seems unnecessary for the test.

Reply:
I took this as they were also present in the other repositories; they're being tested in the combinations for metadata I believe. cc @thozza.

Reply:
Disclaimer: I don't know the answer.

From what I learned about repository metadata, the `filelists` contain information about the content (files and directories) of the packages in the repository. So, not including it may, in theory, work for basic depsolving. However, we can't remove it from all repositories, because the information about provided files and dependencies on them is used to determine dependencies between packages when generating SBOMs. This is also done in our tests.

So removing `filelists` may be OK, but it may also impact SBOM testing. However, it is theoretically possible to delete them from the newly added appstream repo (I think), but no guarantees.

Reply:
I would really like us to explore how to avoid embedding 18mb compressed binary data into our git tree for this new test. For filelists.xml.gz that is already in the tree it is already in the tree, nothing can be done. Maybe there is a stable URL to download it from (a lot of our tests require downloading/caching stuff and that would be an incrediable cheap/simple option)? I can also explore if pruning it/removing it is possible, a quick way to find out would be to just delete it, remove it from the repomd.xml and see if this one new particular test still generates the expected results. I can look into this next week if required.

Reply:
The filelists are explicitly tested; if you want to fiddle with removing them or not you can do so. This is now the only thing blocking this PR.

Reply:
I looked at this now (sorry that it took me so long) and I think https://github.com/osbuild/osbuild/commit/b1a4e287146d6c82293d17ee8683f5f9424a7fe2 is a way out and avoids the 18mb by just hand-crafting an empty filelist. 

Please have a look It will need to be squashed into the commit that adds the filelist (with an explanation about the "why") so that we never have the 18mb in the git history. I did not test with dnf5, it may need tweaks there (dnf4 is pretty liberal with the checking, if e.g. open checksum does not match it does not care on my dnf4 version).

Reply:
My only worry is that this may affect the SBOM tests, but from what I can tell, we are not explicitly testing dependencies on files provided by a package, so this approach will probably be acceptable. Specifically, this piece of the code https://github.com/osbuild/osbuild/blob/1a529d78a97c6f661f3ecad46b75a696ae0ed379/osbuild/util/sbom/dnf.py#L101-L103

Reply:
Note that my patch only removed the appstream filelist, we can add a comment in the test that appstream packaes have an empty filelist. But this should not affect anything existing and we can keep the existing filelists as removing them does not really help anything (they are already part of the git history)

Reply:
Don't get me wrong, I'm all for removing it if it does not break existing tests...